scripts/communityModules/VRBuild/VRBuildGrabscale.js

```diff
@@ -0,0 +1,484 @@
+    Entities.editEntity(gizmoCenterID,{
+        position: gizmoPos,        
+        rotation: Quat.multiply(gizmoRot,tempRotation)
+    });
+}
+
+function updateGizmoScale(scaleUpdateX,scaleUpdateY,scaleUpdateZ) {
+    Entities.editEntity(gizmoXaxisID,{        
+        dimensions: {x: GIZMO_CYLINDER_DIAMETER,y: GIZMO_CYLINDER_LENGTH*scaleUpdateX*2,z: GIZMO_CYLINDER_DIAMETER}        
+    });
+    Entities.editEntity(gizmoYaxisID,{        
+        dimensions: {x: GIZMO_CYLINDER_DIAMETER,y: GIZMO_CYLINDER_LENGTH*scaleUpdateY*2,z: GIZMO_CYLINDER_DIAMETER}       
+    });
+    Entities.editEntity(gizmoZaxisID,{        
+        dimensions: {x: GIZMO_CYLINDER_DIAMETER,y: GIZMO_CYLINDER_LENGTH*scaleUpdateZ*2,z: GIZMO_CYLINDER_DIAMETER}       
+    });
+    Entities.editEntity(gizmoXplusID,{
+        localPosition: {x: 0,y: GIZMO_END_POSITION*scaleUpdateX*2,z: 0}       
+    });
+    Entities.editEntity(gizmoXminusID,{
+        localPosition: {x: 0,y: -GIZMO_END_POSITION*scaleUpdateX*2,z: 0}       
+    });
+    Entities.editEntity(gizmoYplusID,{
+        localPosition: {x: 0,y: GIZMO_END_POSITION*scaleUpdateY*2,z: 0}       
+    });
+    Entities.editEntity(gizmoYminusID,{
+        localPosition: {x: 0,y: -GIZMO_END_POSITION*scaleUpdateY*2,z: 0}       
+    });
+    Entities.editEntity(gizmoZplusID,{
+        localPosition: {x: 0,y: GIZMO_END_POSITION*scaleUpdateZ*2,z: 0}       
+    });
+    Entities.editEntity(gizmoZminusID,{
+        localPosition: {x: 0,y: -GIZMO_END_POSITION*scaleUpdateZ*2,z: 0}       
+    });   
+}
+
+function hideGizmo() {
+    for (var i in gizmoShapeIDs) {
+        Entities.editEntity(gizmoShapeIDs[i],{visible: false});
+    }
+}
+
+function showGizmo() {
+    for (var i in gizmoShapeIDs) {
+        Entities.editEntity(gizmoShapeIDs[i],{visible: true});
+    }
+}
+
+function checkIfEntityIsInGizmo(checkID) {
+    var check = false;       
+    for (var i in gizmoShapeIDs) {          
+        if (gizmoShapeIDs[i] === checkID) {
+            check = true;
+        }
+    }           
+    return check;            
+}
+
+function startScaling() {    
+    if (!isScaling) {        
+        showGizmo();
+        var startHandPositionLeft = MyAvatar.getJointPosition(LEFT_JOINT);
+        var startHandPositionRight = MyAvatar.getJointPosition(RIGHT_JOINT);
+        startHandDistance = Vec3.distance(startHandPositionLeft,startHandPositionRight);
+        isScaling = true;
+        oldDimensions = Entities.getEntityProperties(entityToBeGrabbedID,["dimensions"]).dimensions;        
+    }
+    var handDistance = Vec3.distance(handPositionLeft,handPositionRight);
+    newScale = handDistance / startHandDistance;
+    var scaleX = 1;
+    var scaleY = 1;
+    var scaleZ = 1;
+    if (isScaleModeX) {
+        newDimensions = {x: oldDimensions.x * newScale ,y: oldDimensions.y,z: oldDimensions.z};
+        scaleX = newScale;
+        scaleY = 1;
+        scaleZ = 1;
```

Comment:
Don't need these lines because these variables are already set to `1` above. Similarly, elsewhere.